# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindowInner.h
# Commit: 2fc9a3d0ad8b
# Full Hash: 2fc9a3d0ad8bdbce325df78fcd731e925245afaf
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 1 - CookieStore API - WebIDL, r=smaug,edgul
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215144
# ==============================================================================

diff -r 2d598be5ec67 -r 2fc9a3d0ad8b dom/base/nsGlobalWindowInner.h
--- a/dom/base/nsGlobalWindowInner.h	Wed Sep 11 16:09:37 2024 +0000
+++ b/dom/base/nsGlobalWindowInner.h	Wed Sep 11 16:26:54 2024 +0000
@@ -104,6 +104,7 @@
 class Credential;
 class ClientSource;
 class Console;
+class CookieStore;
 class Crypto;
 class CustomElementRegistry;
 class DataTransfer;
@@ -654,6 +655,8 @@
   already_AddRefed<mozilla::dom::Console> GetConsole(JSContext* aCx,
                                                      mozilla::ErrorResult& aRv);
 
+  already_AddRefed<mozilla::dom::CookieStore> CookieStore();
+
   // https://w3c.github.io/webappsec-secure-contexts/#dom-window-issecurecontext
   bool IsSecureContext() const;
 
@@ -1374,6 +1377,7 @@
   RefPtr<mozilla::dom::Crypto> mCrypto;
   RefPtr<mozilla::dom::cache::CacheStorage> mCacheStorage;
   RefPtr<mozilla::dom::Console> mConsole;
+  RefPtr<mozilla::dom::CookieStore> mCookieStore;
   RefPtr<mozilla::dom::Worklet> mPaintWorklet;
   RefPtr<mozilla::dom::External> mExternal;
   RefPtr<mozilla::dom::InstallTriggerImpl> mInstallTrigger;